### PR TITLE
add trigger sdk

### PR DESCRIPTION
**Pull request by CiCi503**
add trigger sdk

### CODE REVIEW

File: test/trigger_test.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                                       trigger_config, source_arn, invocation_role)
+
+        # get
+        # 200 ok
+        logging.info('get trigger: {0}'.format(trigger_name))
+        get_trigger_resp = self.client.get_trigger(service_name, function_name, trigger_name).data
+        self.check_trigger_response(get_trigger_resp, trigger_name, trigger_type, trigger_config, source_arn,
+                                    invocation_role)
+        # 404
+        with self.assertRaises(fc2.FcError):
```
Comment (rsonghuster) [inline test/trigger_test.py]: 这里可以再加一个invalid function导致404的

File: test/trigger_test.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            self.client.delete_trigger(service_name, function_name, trigger_name)
+
+    def check_trigger_response(self, resp, trigger_name, trigger_type, trigger_config, source_arn, invocation_role):
+        self.assertEqual(resp['triggerName'], trigger_name)
+        self.assertEqual(resp['triggerType'], trigger_type)
+        self.assertEqual(resp['sourceArn'], source_arn)
+        self.assertEqual(resp['invocationRole'], invocation_role)
+        self.assertTrue('createdTime' in resp)
+        self.assertTrue('lastModifiedTime' in resp)
+        self.assertTrue('triggerConfig' in resp)
```
Comment (rsonghuster) [inline test/trigger_test.py]: 这个'triggerConfig'是字符串，函数check_trigger_response的参数trigger_config没有用上？

File: fc2/client.py [hunk 3]
```diff
@@ -575,6 +575,160 @@ def invoke_function(self, serviceName, functionName, payload=None, headers = {})
...
         return FcHttpResponse(r.headers, r.content)
 
+    def create_trigger(self, serviceName, functionName, triggerName, triggerType, triggerConfig, sourceArn, invocationRole, headers={}):
+        """
+        Create a trigger.
+        :param serviceName: (required, string), name of the service that the trigger belongs to.
+        :param functionName: (required, string), name of the function that the trigger belongs to.
+        :param triggerName: (required, string), name of the trigger.
+        :param triggerType: (required, string), the type of trigger. 'oss','log','timer'
+        :param triggerConfig: (required, string), the config of the trigger, different types of trigger has different config.
```
Comment (wanghq) [inline fc2/client.py]: triggerConfig should be dict, not string type.

File: fc2/client.py [hunk 4]
```diff
@@ -575,6 +575,160 @@ def invoke_function(self, serviceName, functionName, payload=None, headers = {})
...
+        :param sourceArn: (required, string), Aliyun Resource Name（ARN）of the event.
```
Comment (wanghq) [inline fc2/client.py]: no longer required

File: fc2/client.py [hunk 5]
```diff
@@ -575,6 +575,160 @@ def invoke_function(self, serviceName, functionName, payload=None, headers = {})
...
+        :param invocationRole: (required, string), the role that event source uses to invoke the function.
```
Comment (wanghq) [inline fc2/client.py]: not required for all triggers, e.g. time trigger

File: fc2/client.py [hunk 6]
```diff
@@ -575,6 +575,160 @@ def invoke_function(self, serviceName, functionName, payload=None, headers = {})
...
+        headers = self._build_common_headers(method, path, headers)
+        self._do_request(method, path, headers)
+
+    def update_trigger(self, serviceName, functionName, triggerName, triggerConfig=None, invocationRole=None, headers={}):
+        """
+        Update a trigger.
+        :param serviceName: (required, string), name of the service that the trigger belongs to.
+        :param functionName: (required, string), name of the function that the trigger belongs to.
+        :param triggerName: (required, string), name of the trigger.
+        :param triggerConfig: (optional, string), the config of the trigger, different types of trigger has different config.
```
Comment (wanghq) [inline fc2/client.py]: same as above

File: fc2/client.py [hunk 7]
```diff
@@ -575,6 +575,160 @@ def invoke_function(self, serviceName, functionName, payload=None, headers = {})
...
+        :param invocationRole: (optional, string), the role that event source uses to invoke the function.
```
Comment (wanghq) [inline fc2/client.py]: same as above

**Pull request merged**